Datadog/E2ETests/Tracing/SpanE2ETests.swift

```diff
@@ -0,0 +1,181 @@
+    /// $monitor_query = "sum(last_1d):avg:trace.trace_span_set_operation_name.hits{service:com.datadog.ios.nightly,env:instrumentation}.as_count() < 1"
+    /// $monitor_threshold = 1
+    /// ```
+    ///
+    /// - performance monitor:
+    /// ```apm
+    /// $feature = trace
+    /// $monitor_id = trace_span_set_operation_name_performance
+    /// $monitor_name = "[RUM] [iOS] Nightly Performance - trace_span_set_operation_name: has a high average execution time"
+    /// $monitor_query = "avg(last_1d):p50:trace.perf_measure{env:instrumentation,resource_name:trace_span_set_operation_name,service:com.datadog.ios.nightly} > 0.024"
+    /// ```
+    func test_trace_span_set_operation_name() {
+        let span = Global.sharedTracer.startSpan(operationName: .mockRandom())
+        let knownOperationName = "trace_span_set_operation_name_new_operation_name" // asserted in monitor
+
+        measure(resourceName: DD.PerfSpanName.fromCurrentMethodName()) {
+            span.setOperationName(knownOperationName)
+        }
+
+        span.finish()
+    }
+
+    /// - api-surface: OTSpan.setTag(key: String, value: Encodable)
+    ///
+    /// - data monitor: (it uses `ios_trace_span_set_tag` metric defined in "APM > Generate Metrics > Custom Span Metrics")
+    /// ```apm
+    /// $feature = trace
+    /// $monitor_id = trace_span_set_tag_data
+    /// $monitor_name = "[RUM] [iOS] Nightly - trace_span_set_tag: number of hits is below expected value"
+    /// $monitor_query = "sum(last_1d):avg:ios_trace_span_set_tag.hits_with_proper_payload{*}.as_count() < 1"
+    /// $monitor_threshold = 1
+    /// ```
+    ///
+    /// - performance monitor:
+    /// ```apm
+    /// $feature = trace
+    /// $monitor_id = trace_span_set_tag_performance
+    /// $monitor_name = "[RUM] [iOS] Nightly Performance - trace_span_set_tag: has a high average execution time"
+    /// $monitor_query = "avg(last_1d):p50:trace.perf_measure{env:instrumentation,resource_name:trace_span_set_tag,service:com.datadog.ios.nightly} > 0.024"
+    /// ```
+    func test_trace_span_set_tag() {
+        let span = Global.sharedTracer.startSpan(operationName: "ios_trace_span_set_tag")
+        let knownTag = DD.specialTag()
+
+        measure(resourceName: DD.PerfSpanName.fromCurrentMethodName()) {
+            span.setTag(key: knownTag.key, value: knownTag.value)
+        }
+
+        span.finish()
+    }
+
+    /// - api-surface: OTSpan.setBaggageItem(key: String, value: String)
+    ///
+    /// - data monitor: (it uses `ios_trace_span_set_baggage_item` metric defined in "APM > Generate Metrics > Custom Span Metrics")
+    /// ```apm
+    /// $feature = trace
+    /// $monitor_id = trace_span_set_baggage_item_data
+    /// $monitor_name = "[RUM] [iOS] Nightly - trace_span_set_baggage_item: number of hits is below expected value"
+    /// $monitor_query = "sum(last_1d):avg:ios_trace_span_set_baggage_item.hits_with_proper_payload{*}.as_count() < 1"
+    /// $monitor_threshold = 1
+    /// ```
```

Comment:
This kind of monitors require manual management of custom span metric generated at ingestion
<img width="1336" alt="Screenshot 2021-09-29 at 13 51 53" src="https://user-images.githubusercontent.com/2358722/135263098-7067e6c5-435f-4ecb-b661-c95bb2433d32.png">

This can't be automated and this is the only way possible if we want to make monitor assertions on span tags.